spec/GraphQLOverHTTP.md

```diff
@@ -112,72 +178,152 @@ http://example.com/product/graphql
-* {operationName} - (*Optional*): The name of the Operation in the Document to execute.
-* {variables} - (*Optional*): Values for any Variables defined by the Operation.
-* {extensions} - (*Optional*): This entry is reserved for implementors to extend the protocol however they see fit.
+Note: {variables} and {extensions}, if set, must have a map as their value.
 
-Note: Be aware that `query` is a misleading name as it can contain a string describing multiple operations, each of which may be a query, mutation or subscription. A better name would have been `document`, but the term `query` is well established.
+## Accept
 
-Note: depending on the serialization format used, values of the aforementioned parameters can be
-encoded differently but their names and semantics must stay the same.
+A client SHOULD indicate the media types that it supports in responses using the
+`Accept` HTTP header as specified in
+[RFC7231](https://datatracker.ietf.org/doc/html/rfc7231).
 
-Note: specifying `null` in JSON (or equivalent values in other formats) as values for optional request parameters is equivalent to not specifying them at all.
+A client SHOULD include `application/graphql+json` in the `Accept` header.
 
-Note: {variables} and {extensions}, if set, must have a map as its value.
+A client MUST include `application/json` in the `Accept` header.
+
+It is RECOMMENDED that a client set the `Accept` header to
+`application/graphql+json; charset=utf-8, application/json; charset=utf-8`.
+
+Note: This recommended header enables compatibility with legacy servers whilst
+still leveraging modern features if available in the server.
 
 ## GET
 
-For HTTP GET requests, the query parameters MUST be provided in the query component of the request URL in the form of
-`key=value` pairs with `&` symbol as a separator and both the key and value should have their "reserved" characters percent-encoded as specified in [section 2 of RFC3986](https://tools.ietf.org/html/rfc3986#section-2).
-The `variables` parameter, if used, MUST be represented as a URL-encoded JSON string.
+For HTTP GET requests, the GraphQL request parameters MUST be provided in the
+query component of the request URL in the `application/x-www-form-urlencoded`
+format as specified by
+[WhatWG's URLSearchParams class](https://url.spec.whatwg.org/#interface-urlsearchparams).
+
+The `query` parameter MUST be the string representation of the Source Text of
+the Document as specified in
+[the Language section of the GraphQL specification](https://spec.graphql.org/draft/#sec-Language).
+
+The `variables` parameter, if used, MUST be represented as a URL-encoded JSON
+string.
 
 ### Example
 
 If we wanted to execute the following GraphQL query:
 
-```graphql example
-query ($id: ID!) {
-  user(id:$id) {
-    name
-  }
-}
+```raw graphql example
+query($id: ID!){user(id:$id){name}}
 ```
 
 With the following query variables:
 
 ```json example
```

Comment:
Should this also be in the raw encoded form?

Reply:
No; the whitespace in the JSON is unimportant.

Reply:
Right, but that is also true for whitespace in the GraphQL. I think it would be self-consistent if both examples were either minified or formatted.

Reply:
You're right; the encoding in the URL should match that in the example :+1: